packages/selection/src/containers/SelectionContainer.spec.js

```diff
@@ -0,0 +1,493 @@
+
+jest.useFakeTimers();
+
+describe('SelectionContainer', () => {
+  const itemValues = ['Item-1', 'Item-2', 'Item-3'];
+  let wrapper;
+
+  const BasicExample = ({ direction, defaultFocusedIndex, selectedAriaKey } = {}) => (
+    <SelectionContainer direction={direction} defaultFocusedIndex={defaultFocusedIndex}>
+      {({ getContainerProps, getItemProps, focusedItem, selectedItem }) => (
+        <div {...getContainerProps({ 'data-test-id': 'container' })}>
+          {itemValues.map(item => {
+            const ref = React.createRef();
+            const isSelected = item === selectedItem;
+            const isFocused = item === focusedItem;
+
+            return (
+              <div
+                {...getItemProps(
+                  {
+                    key: item,
+                    item,
+                    focusRef: ref,
+                    ref,
+                    selectedAriaKey,
+                    'data-test-id': 'item',
+                    'data-focused': isFocused,
+                    'data-selected': isSelected
+                  },
+                  {
+                    selectedAriaKey
+                  }
+                )}
+              >
+                {item}
+              </div>
+            );
+          })}
+        </div>
+      )}
+    </SelectionContainer>
+  );
+
+  const findItems = enzymeWrapper => enzymeWrapper.find('[data-test-id="item"]');
+  const findContainer = enzymeWrapper => enzymeWrapper.find('[data-test-id="container"]');
+
+  beforeEach(() => {
+    act(() => {
+      wrapper = mount(<BasicExample />);
+    });
+  });
+
+  describe('getContainerProps', () => {
+    it('applies accessibility role', () => {
+      const container = findContainer(wrapper);
+
+      expect(container).toHaveProp('role', 'listbox');
+    });
+
+    it('first item in container defaults as the only initial focusable item', () => {
+      let item = findItems(wrapper).first();
+
+      item.simulate('focus');
+
+      // Need to requery item to get update dom attribute
+      item = findItems(wrapper).first();
+      expect(item).toHaveProp('tabIndex', 0);
+      expect(item).toHaveProp('data-focused', true);
+    });
+
+    describe('onFocus', () => {
+      it('does not focus item if item is moused down', () => {
+        findItems(wrapper)
+          .first()
+          .simulate('click');
+
+        expect(findItems(wrapper).first()).toHaveProp('data-focused', false);
+        expect(findItems(wrapper).first()).toHaveProp('data-selected', true);
+      });
+
+      it('focuses first item if no item is currently selected', () => {
+        findItems(wrapper)
+          .first()
+          .simulate('focus');
+
+        expect(findItems(wrapper).first()).toHaveProp('data-focused', true);
+      });
+
+      it.skip('focuses last item if no item is currently selected and defaultFocusedIndex is provided', () => {
+        wrapper = mount(<BasicExample defaultFocusedIndex={-1} />);
```

Comment:
Currently skipping this as we don't currently support `defaultFocusedIndex` in the `useSelection` hook, should we support this usecase where the last item is the initial item to apply `tabIndex: 0` if no controlled state is set?

Reply:
I'm not sure. This really only makes sense in an "uncontrolled" scenario, but is trivial if they just want to see the last item as the current `focusedItem`. We might be able to avoid this and reduce the complexity.

Reply:
Supporting the `-1` shorthand option is going to be tricky but we could just not do that and if they wanted to do it they could just do:

```jsx
const Selection = () => {
  const items = [];
  const { ...props } = useSelection({ defaultFocusedIndex: items.length - 1})
}
``` 

I think that's a good compromise, thoughts?

Reply:
Support for defaultFocusedIndex is here 564e1ca